src/main/java/hudson/plugins/ansicolor/ColorConsoleAnnotator.java

```diff
@@ -85,7 +83,14 @@ public void emitHtml(String html) {
             CountingOutputStream incoming = new CountingOutputStream(new AnsiHtmlOutputStream(outgoing, colorMap, emitter));
             emitter.incoming = incoming;
             try {
-                byte[] data = s.getBytes(charset);
+                /*
+                 * We only use AnsiHtmlOutputStream for its calls to Emitter.emitHtml when it encounters ANSI escape
+                 * sequences. The output of the stream will be discarded. We encode the String using US_ASCII because
+                 * all ANSI escape sequences can be represented in ASCII using a single byte, and any non-ASCII
+                 * characters will be replaced with '?', which is good, since it means that byte offsets will match
```

Comment:
Sorry for commenting out of nowhere, but will that work with non-BMP characters, like emojis?
In Java, `"😀".length()` is 2 but `"😀".getBytes("US-ASCII").length` seems to be 1.

Reply:
Ok, non-BMP characters were broken by the earlier method because markup text was getting inserted in between the two code units of a surrogate pair in some cases. After ffb5045 they should work correctly. Here's a screenshot of running the following pipeline with that commit:

```
ansiColor {
    echo '\033[94;1m[ INFO ] 😀😀Wooh\033[0m😀NotCoolored'
} 
```
<img width="380" alt="screen shot 2018-12-12 at 10 36 37" src="https://user-images.githubusercontent.com/1068968/49880475-3ead5b80-fdfa-11e8-87cf-06008b5f4312.png">

Thanks for the comment!